### PR TITLE
=htc #16472 apply max-content-length setting to Chunked and CloseDelimited entities as well

### LINKED ISSUE #16472: Apply `max-content-length` setting also to chunked encoding
If we keep this setting for the purpose of guarding against `toStrict` blowing up the heap, then we must seal all ways into that particular hell. If we don’t, then we must apply this setting _only_ when calling `toStrict`.

### PR DESCRIPTION
**Pull request by jrudolph**
Fixes #16472.

/cc @sirthias

### CODE REVIEW

**Comment by jrudolph** [comment]
Error lines in https://jenkins.akka.io/job/pr-validator-per-commit-jenkins/3268/:

```
  9950 Test akka.stream.javadsl.FlowTest.mustBeAbleToRecover failed: java.lang.AssertionError: assertion failed: timeout (3 seconds) during expectMsg while waiting for 1, took 3.046 sec
  9951     at scala.Predef$.assert(Predef.scala:179)
  9952     at akka.testkit.TestKitBase$class.expectMsg_internal(TestKit.scala:338)
  9953     at akka.testkit.TestKitBase$class.expectMsg(TestKit.scala:315)
  9954     at akka.testkit.TestKit.expectMsg(TestKit.scala:718)
  9955     at akka.testkit.JavaTestKit.expectMsgEquals(JavaTestKit.java:378)
  9956     at akka.stream.javadsl.FlowTest.mustBeAbleToRecover(FlowTest.java:548)
  9957     ...
 10693 Failed: Total 867, Failed 2, Errors 0, Passed 865, Pending 22
 10694 Failed tests:
 10695 	akka.stream.scaladsl.SinkForeachParallelSpec
 10696 	akka.stream.javadsl.FlowTest
 14047 (akka-stream-tests-experimental/test:test) sbt.TestsFailedException: Tests unsuccessful
 14048 Total time: 925 s, completed Jul 21, 2015 6:38:55 PM
```

File: akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpRequestParser.scala [hunk 1]
```diff
@@ -187,10 +188,16 @@ private[http] class HttpRequestParser(_settings: ParserSettings,
           if (te.isChunked) {
             if (clh.isEmpty) {
               emitRequestStart(chunkedEntity(cth, expect100continueHandling), completedHeaders)
-              parseChunk(input, bodyStart, closeAfterResponseCompletion)
+              parseChunk(input, bodyStart, closeAfterResponseCompletion, totalBytesRead = 0L)
```
Comment (2beaucoup) [inline akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpRequestParser.scala]: I'd switch to a default param for of 0 for totalBytesRead.

Comment (jrudolph) [inline akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpRequestParser.scala]: I'm a bit torn about this pattern as it makes it so easy to forget to provide it for the recursive calls.

Comment (2beaucoup) [inline akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpRequestParser.scala]: I agree. It boils down to taking a bit more care on the inside vs. making it easier to use from the outside. If it's used solely locally I'd also vote against defaults.

**Comment by jrudolph** [comment]
I think this could be merged.

**Comment by jrudolph** [comment]
PLS BUILD

**Comment by ktoso** [comment]
LGTM :)

**Comment by jrudolph** [comment]
Error lines in https://jenkins.akka.io/job/pr-validator-per-commit-jenkins/3319/:

```
 11472 Test akka.stream.javadsl.FlowTest.mustBeAbleToRecover failed: java.lang.AssertionError: assertion failed: timeout (3 seconds) during expectMsg while waiting for 1, took 3.021 sec
 11473     at scala.Predef$.assert(Predef.scala:179)
 11474     at akka.testkit.TestKitBase$class.expectMsg_internal(TestKit.scala:338)
 11475     at akka.testkit.TestKitBase$class.expectMsg(TestKit.scala:315)
 11476     at akka.testkit.TestKit.expectMsg(TestKit.scala:718)
 11477     at akka.testkit.JavaTestKit.expectMsgEquals(JavaTestKit.java:378)
 11478     at akka.stream.javadsl.FlowTest.mustBeAbleToRecover(FlowTest.java:548)
 11479     ...
 12556 Failed: Total 867, Failed 1, Errors 0, Passed 866, Pending 22
 12557 Failed tests:
 12558 	akka.stream.javadsl.FlowTest
 15922 (akka-stream-tests-experimental/test:test) sbt.TestsFailedException: Tests unsuccessful
 15923 Total time: 864 s, completed Aug 4, 2015 11:50:25 AM
```

**Comment by jrudolph** [comment]
Still the same unrelated test failure.

**Pull request merged**